scripts/opencontrail-install/contrail_install.rb

```diff
@@ -165,24 +333,86 @@ def provision_contrail_compute_kubernetes
         fp.puts plugin_conf
     }
 
-    sh(%{sed -i 's/DAEMON_ARGS=" /DAEMON_ARGS=" --network_plugin=#{plugin} /' /etc/sysconfig/kubelet})
-    sh("systemctl restart kubelet")
+    if @platform =~ /fedora/
+        sh(%{sed -i 's/DAEMON_ARGS=" /DAEMON_ARGS=" --network_plugin=#{plugin} /' /etc/sysconfig/kubelet})
+        sh("systemctl restart kubelet", true)
+        sh("systemctl stop kube-proxy", true)
+    else
+        sh(%{sed -i 's/DAEMON_ARGS /DAEMON_ARGS --network_plugin=#{plugin} /' /etc/default/kubelet})
+        sh("service kubelet restart", true)
+
+        # Disable kube-proxy monitoring and stop the service.
+        sh("mv /etc/monit/conf.d/kube-proxy /etc/monit/.", true)
+        sh("monit reload", true)
+        sh("service kube-proxy stop", true)
+    end
+
+    # Flush iptable nat entries
+    sh("iptables -F -t nat")
+end
+
+def aws_setup
+    # Update /etc/hosts
+    # Allow password based login in ssh
+end
+
+def install_kube_network_manager (kubernetes_branch = "release-0.17",
+                                  contrail_branch = "master")
+    ENV["TARGET"]="#{ENV["HOME"]}/contrail"
+    ENV["CONTRAIL_BRANCH"]=contrail_branch
+    ENV["KUBERNETES_BRANCH"]=kubernetes_branch
+    ENV["GOPATH"]="#{ENV["TARGET"]}/kubernetes/Godeps/_workspace"
+    target = @platform =~ /fedora/ ? "/root" : "/home/ubuntu"
+
+    sh("rm -rf #{ENV["TARGET"]}")
+    sh("mkdir -p #{ENV["TARGET"]}")
+    Dir.chdir(ENV["TARGET"])
+
+    commands=<<EOF
+wget -q -O - https://storage.googleapis.com/golang/go1.4.2.linux-amd64.tar.gz | tar -C /usr/local -zx
+rm -rf /usr/bin/go
+ln -sf /usr/local/go/bin/go /usr/bin/go
+git clone -b #{ENV["KUBERNETES_BRANCH"]} https://github.com/googlecloudplatform/kubernetes
+go get github.com/Juniper/contrail-go-api
+wget -q https://raw.githubusercontent.com/Juniper/contrail-controller/#{ENV["CONTRAIL_BRANCH"]}/src/schema/vnc_cfg.xsd
+wget -q https://raw.githubusercontent.com/Juniper/contrail-controller/#{ENV["CONTRAIL_BRANCH"]}/src/schema/loadbalancer.xsd || true
+git clone -b #{ENV["CONTRAIL_BRANCH"]} https://github.com/Juniper/contrail-generateDS.git
+./contrail-generateDS/generateDS.py -f -o ./kubernetes/Godeps/_workspace/src/github.com/Juniper/contrail-go-api/types -g golang-api vnc_cfg.xsd 2>/dev/null
+mkdir -p ./kubernetes/Godeps/_workspace/src/github.com/Juniper/
+ln -sf #{target}/contrail-kubernetes ./kubernetes/Godeps/_workspace/src/github.com/Juniper/contrail-kubernetes
+mkdir -p #{ENV["GOPATH"]}/src/github.com/GoogleCloudPlatform
+ln -sf #{ENV["TARGET"]}/kubernetes #{ENV["GOPATH"]}/src/github.com/GoogleCloudPlatform/kubernetes
+sed -i 's/ClusterIP/PortalIP/' ./kubernetes/Godeps/_workspace/src/github.com/Juniper/contrail-kubernetes/pkg/network/opencontrail/controller.go
+sed -i 's/DeprecatedPublicIPs/PublicIPs/' ./kubernetes/Godeps/_workspace/src/github.com/Juniper/contrail-kubernetes/pkg/network/opencontrail/controller.go
+go build github.com/Juniper/contrail-go-api/cli
```

Comment:
When doing the go build commands set the path to $GOPATH + kubernetes/Godeps/_workspace.
Only a single go build command is required (the kube-network-manager).